src/components/chat/EnhancedChatMessage.tsx

```
@@ -0,0 +1,274 @@
+  copyToClipboard,
+  copiedMessage,
+  onApproveDiagram,
+  onRequestDiagramChanges,
+  isSubmittingDiagram
+}) => {
+  const [showCodeArtifacts, setShowCodeArtifacts] = useState(true);
+  const [isExpanded, setIsExpanded] = useState(false);
+
+  // Extract code blocks from message content
+  const extractCodeBlocks = (content: string) => {
+    const codeBlockRegex = /```(\w+)?\n([\s\S]*?)```/g;
```

ESLint

[error] Unsafe Regular Expression

(security/detect-unsafe-regex)